docs/source/dev_guides/languagebasedtools.rst

```diff
@@ -0,0 +1,138 @@
+.. _languagebasedtools:
+
+====================
+Language-Based Tools
+====================
+
+Language-based tools are software that operate on programming languages,
+including Integrated Development Environments (IDEs) and others.
+
+Many of these tools can be configured to recognise the syntax of a .enaml file,
+and thus provide syntax-colouring, indenting support and other features.
+
+Enaml comes bundled with a number of configuration files for common IDEs and
+other tools.
+
+`BBEdit`_
+----------------------------
+
+BBEdit is a commercial MacOS IDE from `Bare Bones`_.
+
+An Enaml language library for BBEdit is provided in the source at
+``tools/barebones/``
+
+For installation instructions, see the `BBEdit Support pages`_.
+
+.. _BBEdit: https://www.barebones.com/products/bbedit/
+.. _Bare Bones: https://www.barebones.com/
+.. _BBEdit Support pages: https://www.barebones.com/support/bbedit/plugin_library.html
+
+`GNU Emacs`_
+------------
+
+GNU Emacs is a popular, cross-platform, open-source IDE.
+
+An Enaml mode for Emacs is provided in the source at ``tools/emacs``.
+
+For installation instructions, see in the source at
+``tools/emacs/README.rst``
+
+.. _GNU Emacs: https://www.gnu.org/software/emacs/
+
+`TextMate`_
+-----------
+
+TextMate is a commercial MacOS IDE.
+
+An Enaml language definition for TextMate is provided in the source tree at
+``tools/sublimetext``.
+
+
+For installation instructions, see the `TextMate Manual`_.
+
+.. _TextMate: https://macromates.com/
+.. _TextMate Manual: https://macromates.com/manual/en/language_grammars#language_grammars
+
+`Sublime Text`_
+---------------
+
+Sublime Text is a cross-platform, commercial IDE.
+
+Sublime Text can also use the TextMate language definition at
+``tools/sublimetext/Enaml.tmLanguage``.
+
+For installation instructions, see the `Sublime Text manual`_.
+
+.. _Sublime Text: https://www.sublimetext.com/
+.. _Sublime Text manual: http://docs.sublimetext.info/en/latest/extensibility/packages.html#installing-packages
+
+
+`Visual Studio`_
+----------------
+
+Visual Studio is a commercial IDE for Windows and MacOs from Microsoft.
+
+Visual Studio can also use the TextMate language definition at
+``tools/sublimetext/Enaml.tmLanguage``.
```

Comment:
I created a package for visual studio (enaml-vs) that can be installed directly for the extension manager. May be worth mentionning.

Reply:
Done. Thanks. 

[I am counting this as another justification for documenting these features, because I didn't know enaml-vs existed.]